deepfence_server/handler/scan_reports.go

```diff
@@ -1877,23 +1922,32 @@ func StartMultiScan(ctx context.Context,
 			[]model.NodeIdentifier{},
 			req.Filters)
 		if err != nil {
-			return nil, "", err
+			return nil, "", skip_node_ids, err
 		}
 		reqs = append(reqs, reqs_extra...)
 	} else {
 		reqs = req.NodeIds
 	}
 
+	// check and skip if scan are already running or scheduled
+	for _, n := range reqs {
+		if IsScanInProgress(ctx, scan_type, n) {
```

Comment:
I think it will be better to get all currently running scans in one query and compare it against the map.

Reply:
If I want to query all node_ids it will be a where query and might take longer to complete, this is a simple read query should be faster and ok to run multiple times, also as of now we can start scans on 50nodes at at time so this should be ok

Reply:
That's a tricky one, I agree with Ramanan, having 1 read instead of N is preferable, we just need to replace:
```
Match (n:%s{node_id:$node_id})
```
with
```
MATCH (n:%s) WHERE n.node IN $all_ids
```

Now I am not a big fan of relying on a map in memory to determine if we should start a scan or not. We could handle that in the start scan neo4j query itself. It used to be done like that (and I am surprised to see this was removed over time, we should have it in the git history?)
The way it used to work is: we were starting all scans and if the scan was IN_PROGRESS, we were passing the old in_progress scan_id as if it was started.
It could be done via a combination of `OPTIONAL MATCH` and `MERGE`